### ISSUE TITLE
Auto fan curve going to min fan speed under load (and temps close to the load temp)

### ISSUE THREAD

**Issue by byKaLeD** [Rem0o/FanControl.Releases#1435]
Right after I started using Fan Control I began to have crashes after 15-20 mins of gaming, sometimes followed by my PC not booting and showing a DRAM error that was solved by cooling down my RAM modules. It got to the point where I had to reinstall windows since it took me some time to diagnose it and it had already started to corrupt my OS.

After a clean OS install, I wanted to know why my RAM was getting so heat soaked that it would get unstable and crash (at stock XMP setting btw). So I imported my Fan Control settings and opened the UI on my second monitor and after like 10 mins I saw that 2 of my 3 auto fan curves were at min fan speed while the temps on the source were very high. +75° on the CPU, +95° Junction GPU and 49° inside the case. This happens to all 3 of my auto fan curves, sometimes one of them at a time and sometimes multiple (or all) at the same time.

Since then, I've tried changing every setting on Fan Control, BIOS settings, putting the program on high priority, and changing the temperature source but nothing has solved the problem. The only pattern I found is that this happens when I have 100% (or close to it) GPU usage for some period of time (I've seen it after as little as 2 mins but is usually a bit more). In fact right as I close the game or I put a framerate cap below 99% GPU usage, everything works as Intended. I've found that changing a setting on the auto fan curve or refreshing the sensors temporarily fixes the problem but it doesn't last long before it happens again.

TL;DR After some time of high load on the GPU, auto fan curves go to the min fan speed and stay there as temperatures keep raising and causing my PC to heat soak and crash.

Screenshot of Fan Control with my PC at idle
![Screenshot_20230106_113840](https://user-images.githubusercontent.com/122105025/211053475-aee44638-c10f-4f7c-995f-5d01f9f9ed2c.png)

Screenshot of Fan Control after less than 5 mins of Assetto Corsa
![Screenshot_20230106_115330](https://user-images.githubusercontent.com/122105025/211053659-1e2850c0-ec7a-4893-814b-38f7e6e4baf9.png)

FYI, the case's temperature source (Temperature #.2 - MSI B450...) is a sensor inside the case that is on the MOBO like 3-5cm below the GPU.

Keep in mind that here in the southern hemisphere is summer and my room sees over 33°C for a couple of hours a day pretty much every day.

Relevant PC specs
Windows 11 on the latest version
R5 3600
ID Cooling SE 224 XT Black (push pull setup) with Kryonaut thermal paste
RX 6700 XT Red Devil (undervolted and underclocked so my PC doesn't heat soak so fast)
B450 Tomahawk Max
2x8GB HyperX Predator 3600 MT/s Cl 17 (right now at 3200 MT/s with the XMP timings, so that it doesn't crash so fast, although it still does it after about an hour and a half, even with these veeery conservative settings. But before I had like 20 mins or 30 if it didn't heat up so fast)
DeepCool MATREXX 55 MESH
Intake (3) and top exhaust (2) fans are Artic P14 PWM
CPU (2) and back exhaust (1) fans are some DeepCool 120mm RGB fans that came with my case.

Even though I feel like I already tried everything, any help or idea to fix the problem is appreciated but I do believe that this is a bug on the software and it needs an update to be fixed. But who knows, maybe I'm not seeing something and it's an easy fix.

**Comment by Rem0o** [comment]
The auto fan curve will pull the speed down until the temp starts creeping to/over the load temperature, and will try to find equilibrium.  That's how they work. The load temperature should be expected to be hit. 

From what I can see, you set your load temperatures too high, and your min fan speeds too low. Auto fan curve works great for constant loads, like 24/7 coin mining and stuff like that, where the function will have time to adapt and find the optimal speed for a given load temperature.

Auto fan curve is probably not the right type of fan curve for what you want to achieve. I would recommend replacing them with graphs or linears.

**Comment by byKaLeD** [comment]
> The auto fan curve will pull the speed down until the temp starts creeping to/over the load temperature, and will try to find equilibrium. That's how they work. The load temperature should be expected to be hit.

Based on the description of the auto fan curve I assumed that but I expected the pull-down of speed to be gradual, but from what I've seen (since there's no logging) it goes directly back down to the min, without "checking" other [lower] speeds. Maybe a bug or maybe the function working properly but my load temp is way too high causing that behavior.

**Comment by byKaLeD** [comment]
> The auto fan curve will pull the speed down until the temp starts creeping to/over the load temperature, and will try to find equilibrium. That's how they work. The load temperature should be expected to be hit.

Based on the description of the auto fan curve I assumed that but I expected the pull-down of speed to be gradual, but from what I've seen (since there's no logging) it goes directly back down to the min, without "checking" other [lower] speeds at equilibrium or load temp. Maybe a bug or maybe the function working properly but my load temp is way too high causing that behavior.

> From what I can see, you set your load temperatures too high, and your min fan speeds too low. Auto fan curve works great for constant loads, like 24/7 coin mining and stuff like that, where the function will have time to adapt and find the optimal speed for a given load temperature.

> Auto fan curve is probably not the right type of fan curve for what you want to achieve. I would recommend replacing them with graphs or linears.

Well, I wanted something that would adapt the speeds so I don't have to take the time to create a fan curve for summer, another for winter, and everything in between but I guess this isn't the intended use case and I'm not gonna get the behavior I want of out it.

Thank you for answering so quickly, I appreciate the explanation. Also, should I delete the issue since it isn't really an issue or I should leave it so people with the same problem can check it out?
